lib/resque/failure/extra.rb

```diff
@@ -24,15 +24,36 @@ def matching_job?(job, opts)
         opts.all? { |k, v|
           case k.to_sym
           when :class
-            job['payload']['class']
+            job['payload']['class'] == v
           when :smart
-            smart_classify(job)
+            smart_classify(job) == v
+          when :older_than
+            job_older_than?(job, v)
           else
-            job[k.to_s]
-          end == v
+            job[k.to_s] == v
+          end
         }
       end
 
+      def job_older_than?(job, max_age)
+        now = DateTime.now
+        diff = now - DateTime.parse(job["failed_at"])
```

Comment:
What's the unit of that subtraction? Use that unit in the variable name, eg `diff_days`, so it's easier to follow.